syntax = "proto3";

option java_multiple_files = true;
option java_package = "primary-block-protocol";
option java_outer_classname = "Quorum";
option objc_class_prefix = "DSC";

package primary_block;

service Registry {
    rpc Register(ServerAddress) returns (ServerAddress) {} //return information about primary replica
    rpc GetServerList(EnquireServers) returns (stream ServerAddress) {} 
}

message ServerAddress {
    string ip = 1;
    string port = 2;
}

message EnquireServers {}